Pipeline/PackageBuildOutputs/ArtifactoryHelpers.groovy

```diff
@@ -0,0 +1,182 @@
+@groovy.transform.BaseScript com.ibm.dbb.groovy.ScriptLoader baseScript
+import groovy.transform.*
+
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.entity.FileEntity;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.protocol.HTTP;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.impl.client.DefaultHttpRequestRetryHandler;
+
+// Very basic script to upload/download from artifactory (replacement of curl)
+@Field int retryCount = 50
+@Field int conPoolSize = 10
+@Field int timeout = 300 // 5 Minutes in seconds
+@Field int timeoutMilliSeconds = timeout * 1000;
+
+class CustomRetryHandler extends DefaultHttpRequestRetryHandler {
+	boolean verbose = false
+	CustomRetryHandler(int connectionRetries, boolean verbose) {
+		super(connectionRetries, true);
+		this.verbose = verbose
+	}
+
+	@Override
+	public boolean retryRequest(IOException exception, int totalCount, HttpContext context) {
+		HttpClientContext clientContext = HttpClientContext.adapt(context);
+		if ( verbose) println("*? WARNING: Error occurred for request " + clientContext.getRequest().getRequestLine().toString() + ": " + exception.getMessage() + ".");
+		if (totalCount > retryCount) {
+			println("*? Error occurred for request " + clientContext.getRequest().getRequestLine().toString() + ": " + exception.getMessage() + ".");
+			return false;
+		}
+		Thread.sleep(1000);
+		boolean shouldRetry = super.retryRequest(exception, totalCount, context);
+		if (shouldRetry) {
+			if ( verbose) println("*? WARNING: Attempting retry #" + totalCount);
+			return true;
+		}
+		return false;
+	}
+}
+
+run(args)
+
+def upload(String url, String fileName, String user, String password, boolean verbose) throws IOException {
+
+	System.setProperty("com.ibm.jsse2.overrideDefaultTLS", "true")
+	
```

Comment:
Can you remove this line? 
This is something that should be pass in JAVA Opts (if needed) -Dcom.ibm.jsse2.overrideDefaultTLS=true. In OCP demos I hard-coded it in the groovy script (to not modify Ansible playbooks)..